pom.xml

```diff
@@ -528,94 +542,44 @@
                 <showWarnings>true</showWarnings>
-                <compilerArgs>
-                  <arg>-Xlint:all</arg>
-                </compilerArgs>
-              </configuration>
-            </plugin>
-            <plugin>
-              <groupId>org.apache.maven.plugins</groupId>
-              <artifactId>maven-checkstyle-plugin</artifactId>
-              <version>${checkstyle.version}</version>
-              <executions>
-                <execution>
-                  <id>validate</id>
-                  <phase>validate</phase>
-                  <configuration>
-                    <configLocation>${checkstyle.config.location}</configLocation>
-                    <encoding>UTF-8</encoding>
-                    <consoleOutput>true</consoleOutput>
-                    <!-- Remove or switch to false to keep building even with checkstyle errors -->
-                    <failsOnError>true</failsOnError>
-                  </configuration>
-                  <goals>
-                    <goal>check</goal>
-                  </goals>
-                </execution>
-              </executions>
-            </plugin>
-          </plugins>
-        </pluginManagement>
-      </build>
-    </profile>
-    <profile>
-      <id>jdk8</id>
-      <activation>
-        <jdk>1.8</jdk>
-      </activation>
-      <build>
-        <pluginManagement>
-          <plugins>
-            <plugin>
-              <groupId>org.apache.maven.plugins</groupId>
-              <artifactId>maven-compiler-plugin</artifactId>
-              <version>3.8.0</version>
-              <configuration>
-                <source>1.8</source>
-                <target>1.8</target>
-                <compilerId>javac-with-errorprone</compilerId>
                 <forceJavacCompilerUse>true</forceJavacCompilerUse>
-                <showWarnings>true</showWarnings>
                 <compilerArgs>
+                  <arg>-XDcompilePolicy=simple</arg>
+                  <!--
+                  all flags to errorprone should be passed in the same <arg> as -Xplugin.
+                  -->
+                  <arg>
+                     -Xplugin:ErrorProne
+                     -Xep:RemoveUnusedImports:ERROR
+                     -Xep:PrivateConstructorForUtilityClass:ERROR
+                     -Xep:FieldMissingNullable:ERROR
+                     -Xep:MultipleTopLevelClasses:ERROR
+                     -Xep:MultipleUnaryOperatorsInMethodCall:ERROR
+                     -Xep:PackageLocation:ERROR
+                     -Xep:SwitchDefault:ERROR
+                     -Xep:ThrowsUncheckedException:ERROR
+                     -Xep:TypeParameterNaming:ERROR
+                     -Xep:UngroupedOverloads:WARN
+                     -Xep:UnnecessaryStaticImport:ERROR
+                     -Xep:WildcardImport:ERROR
+                     -XepDisableWarningsInGeneratedCode
+                     -XepExcludedPaths:.*/generated-(test-)?sources/.*
+                     -Xep:Var:OFF
```

Comment:
Why the explicit disable of these particular rule? They are not enabled by default afaik.

Reply:
I thought they are here to enable, not disable. Maybe I misunderstood. @danielnorberg

Reply:
Might be copy pasta from internal root pom

Reply:
OFF means disable

Reply:
Ah OK, I misunderstood the comment. It was only for that line. `these particular rule` confused me. :D

Reply:
That line and the ones below I guess

Reply:
And yes, this was copy-pasta from internal root pom. Not sure why they were disabled there.

Reply:
They were disabled there because of another setting that would enable all non-default checks as warnings and some of them were buggy. It should not be needed if you do not have that setting (and error-prone authors do not recommend that setting).

Reply:
Aha TIL. Thanks